tests/app/views/test_frameworks.py

```diff
@@ -202,62 +200,151 @@ class TestUpdateFramework(BaseApplicationTest, JSONUpdateTestMixin):
+            'frameworkAgreementDetails',
+            'status',
+            'clarificationQuestionsOpen',
+        ]
 
-    def test_framework_updated(self):
+    def post_framework_update(self, update):
+        return self.client.post(
+            '/frameworks/example-framework',
+            data=json.dumps({
+                'frameworks': update,
+                'updated_by': 'example user'
+            }),
+            content_type="application/json"
+        )
+
+    def test_returns_404_on_non_existent_framework(self, open_example_framework):
         with self.app.app_context():
-            response = self.client.post('/frameworks/example',
-                                        data=json.dumps({'frameworks': {
-                                            'status': 'expired',
-                                            'clarificationQuestionsOpen': False,
-                                        }, 'updated_by': 'example user'}),
-                                        content_type="application/json")
+            response = self.client.post(
+                '/frameworks/example-framework-2',
+                data=json.dumps({'frameworks': {
+                    'status': 'expired'
+                }, 'updated_by': 'example user'}),
+                content_type="application/json"
+            )
 
-            assert response.status_code == 200
+            assert response.status_code == 404
 
-            framework = Framework.query.filter(Framework.slug == 'example').first()
-            assert framework.status == "expired"
-            assert not framework.clarification_questions_open
+    def test_can_update_whitelisted_fields(self, open_example_framework):
+        valid_attributes_and_values = {
+            key: value for key, value in self.framework_attributes_and_values_for_update.items()
+            if key in self.attribute_whitelist
+        }
 
-    def test_returns_404_on_non_existent_framework(self):
         with self.app.app_context():
-            response = self.client.post('/frameworks/example2',
-                                        data=json.dumps({'frameworks': {'status': 'expired'},
-                                                         'updated_by': 'example user'}),
-                                        content_type="application/json")
+            for key, value in valid_attributes_and_values.items():
+                response = self.post_framework_update({
+                    key: value
+                })
+
+                assert response.status_code == 200
+                data = json.loads(response.get_data())['frameworks']
+                # frameworkAgreementDetails doesn't show up in the serialized framework
+                if isinstance(value, dict):
+                    for nested_key, nested_value in value.items():
+                        assert data[nested_key] == nested_value
+                else:
+                    assert data[key] == value
```

Comment:
I personally don't think it's good enough to just check the response from a modifying call. I would also want to perform a secondary `GET` and "check it's stuck".

Very easy to add the check - you can probably just compare equality to the `POST`-returned response.